src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,454 @@
+            }
+        });
+
+        return valid;
+    }
+    
+    save() {
+        if (!this.isEntriesValid()) {
+            if (this.viewMode() === "summaryMode") {
+                this.switchToEditMode()
+            }
+
+            return;
+        }
+
+        const settingsToSave = this.allEntries().map(entry => {
+            if (entry instanceof models.databaseEntry) {
+                return entry.getEntrySetting();
+            }
+        }).filter(x => !!x);
+
+        const settingsToSaveSorted = _.sortBy(settingsToSave, x => x.key);
+        
+        const saveSettingsModel = new saveDatabaseSettingsConfirm((settingsToSaveSorted))
+
+        saveSettingsModel.result.done(result => {
+            if (result.can) {
+                this.spinners.save(true);
+
+                new saveDatabaseSettingsCommand(this.activeDatabase(), settingsToSaveSorted)
+                    .execute()
+                    .done(() => {
+                        this.fetchDatabaseSettings(this.activeDatabase())
+                            .done(() => this.processAfterFetch())
+                    })
+                   .always(() => this.spinners.save(false))
+                }
+            });
+
+        app.showBootstrapDialog(saveSettingsModel);
+    }
+
+    refresh() {
+        this.confirmRefresh()
+            .then(result => {
+                if (result.can) {
+                    this.spinners.refresh(true);
+
+                    this.fetchDatabaseSettings(this.activeDatabase(), true)
+                        .done(() => {
+                            this.processAfterFetch();
+                            this.dirtyFlag().reset();
+                        })
+                        .always(() => this.spinners.refresh(false));
+                }
+            });
+    }
+
+    private processAfterFetch() {
+        if (this.viewMode() === "editMode") {
+            this.setupEditMode();
+        } else {
+            this.computeEntriesToShow();
+        }
+    }
+    
+    confirmRefresh() {
+        let text = `<div>Clicking Refresh will reload all database settings from the server.</div>`;
+        if (this.isSaveEnabled()) {
+            text = "<div><span class='bg-warning text-warning'><i class='icon-warning margin-right margin-right-sm'></i>You have unsaved changes !</span></div>" + text;
+        }
+
+        return this.confirmationMessage("Are you sure ?", text,
+            {
+                html: true,
+                buttons: ["Cancel", "Refresh"]
+            });
+    }
+    
+    setGroupToShow(groupType: entriesGroupType) {
+        this.groupToShow(groupType);
+    }
+
+    exitEditMode() {
+        this.viewMode("summaryMode");
+        this.computeEntriesToShow();
+    }
+    
+    switchToEditMode() {
+        if (!this.editModeHasBeenEntered) {
+            this.confirmationMessage("Are you an expert?",
+                "Modify the database settings only if you know what you are doing!", { buttons: ["OK"] })
```

Comment:
I wonder if we should cancel as well - someone may say: I'm not the expert.

Reply:
done